test/procedure.test.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+var Database = require('../lib/database');
+var drivers = require('./drivers');
+var mssql = require('mssql');
+var should = require('should');
+
+var getDriver = function(name, opts) {
+  return new Database(name, opts);
+};
+
+var driver = drivers[1];
+var db = getDriver(driver.name, driver.opts);
+
+before(function(done) {
+  db.execute(driver.dropSP1)
+    .then(function(){
+      db.execute(driver.createSP1);
+    });
+  db.execute(driver.dropSP2)
+    .then(function(){
+      db.execute(driver.createSP2);
+    });
+  db.execute(driver.dropSP3)
+    .then(function(){
+      db.execute(driver.createSP3);
+    });
+  done();
+});
+
+after(function(done) {
+  db.execute(driver.dropSP1);
+  db.execute(driver.dropSP2);
+  db.execute(driver.dropSP3);
+  done();
+});
+
+describe('Stored Procedure', function() {
+
+  it('should return record set', function(done) {
+      db.procedure()
```

Comment:
Unfortunately this isn't going to work, all you're doing here is returning the underlying mssql.Request object, and then your invokation of db.procedure() is simply calling the mssql library.

The purpose of Osql was that you were abstracted away from the underlying driver (mssql, sqlite, mysql etc) and the signiture you use to invoke is always the same.

Therefore, I'd be expecting this to be something like:

```
var opts = {};
opts.args = {
  arg1: value.
  arg2: value
}
db.procedure('Procedure Name, opts);
```
Or something along those lines.

All you've done here is effectively write a bunch of tests that test the mssql module.